frame/support/src/traits/tokens/nonfungible_v2.rs

```diff
@@ -0,0 +1,248 @@
+	///
+	/// By default this is `None`; no attributes are defined.
+	fn attribute(
+		_item: &Self::ItemId,
+		_namespace: &AttributeNamespace<AccountId>,
+		_key: &[u8],
+	) -> Option<Vec<u8>> {
+		None
+	}
+
+	/// Returns the strongly-typed attribute value of `item` corresponding to `key`.
+	///
+	/// By default this just attempts to use `attribute`.
+	fn typed_attribute<K: Encode, V: Decode>(
+		item: &Self::ItemId,
+		namespace: &AttributeNamespace<AccountId>,
+		key: &K,
+	) -> Option<V> {
+		key.using_encoded(|d| Self::attribute(item, namespace, d))
+			.and_then(|v| V::decode(&mut &v[..]).ok())
+	}
+
+	/// Returns `true` if the `item` may be transferred.
+	///
+	/// Default implementation is that all items are transferable.
+	fn can_transfer(_item: &Self::ItemId) -> bool {
+		true
+	}
+}
+
+/// Interface for enumerating items in existence or owned by a given account over a collection
+/// of NFTs.
+pub trait InspectEnumerable<AccountId>: Inspect<AccountId> {
+	/// The iterator type for [`Self::items`].
+	type ItemsIterator: Iterator<Item = Self::ItemId>;
+	/// The iterator type for [`Self::owned`].
+	type OwnedIterator: Iterator<Item = Self::ItemId>;
+
+	/// Returns an iterator of the items within a `collection` in existence.
+	fn items() -> Self::ItemsIterator;
+
+	/// Returns an iterator of the items of all collections owned by `who`.
+	fn owned(who: &AccountId) -> Self::OwnedIterator;
+}
+
+/// Trait for providing an interface for NFT-like items which may be minted, burned and/or have
+/// attributes set on them.
+pub trait Mutate<AccountId, ItemConfig>: Inspect<AccountId> {
+	/// Mint some `item` to be owned by `who`.
+	///
+	/// By default, this is not a supported operation.
+	fn mint_into(
+		_item: &Self::ItemId,
+		_who: &AccountId,
+		_config: &ItemConfig,
+		_deposit_collection_owner: bool,
+	) -> DispatchResult {
+		Err(TokenError::Unsupported.into())
+	}
+
+	/// Burn some `item`.
+	///
+	/// By default, this is not a supported operation.
+	fn burn(_item: &Self::ItemId, _maybe_check_owner: Option<&AccountId>) -> DispatchResult {
+		Err(TokenError::Unsupported.into())
+	}
+
+	/// Set attribute `value` of `item`'s `key`.
+	///
+	/// By default, this is not a supported operation.
+	fn set_attribute(_item: &Self::ItemId, _key: &[u8], _value: &[u8]) -> DispatchResult {
+		Err(TokenError::Unsupported.into())
+	}
+
+	/// Attempt to set the strongly-typed attribute `value` of `item`'s `key`.
+	///
+	/// By default this just attempts to use `set_attribute`.
+	fn set_typed_attribute<K: Encode, V: Encode>(
+		item: &Self::ItemId,
+		key: &K,
+		value: &V,
+	) -> DispatchResult {
+		key.using_encoded(|k| value.using_encoded(|v| Self::set_attribute(item, k, v)))
+	}
+}
+
+/// Trait for providing a non-fungible set of items which can only be transferred.
+pub trait Transfer<AccountId>: Inspect<AccountId> {
+	/// Transfer `item` into `destination` account.
+	fn transfer(item: &Self::ItemId, destination: &AccountId) -> DispatchResult;
+}
```

Comment:
What do you mean "can only be transferred"? Does that mean they could not have any attributes set on them or anything else?

Reply:
This needs to be re-phrased, I assume :)

Reply:
It's the "only" that is the problem for me. Because you could have some collection that is `Transfer + SomethingElse`. It also looks like it takes one `item` and not a `set of items`. So why not just, "Trait for transferring a non-fungible item."?